agent/xds/clusters.go

```diff
@@ -197,9 +207,9 @@ func (s *Server) clustersFromServicesAndResolvers(
 		var cluster *envoy.Cluster
 		var err error
 		if hasResolver {
-			cluster, err = s.makeGatewayClusterWithConnectTimeout(clusterName, cfgSnap, resolver.ConnectTimeout)
+			cluster, err = s.makeGatewayClusterWithConnectTimeout(cfgSnap, clusterName, &svc, resolver.ConnectTimeout)
 		} else {
-			cluster, err = s.makeGatewayCluster(clusterName, cfgSnap)
+			cluster, err = s.makeGatewayCluster(cfgSnap, clusterName, &svc)
```

Comment:
Not directly related to your change, but in reference to my comment about making 3 calls into 2:

I think this could be made to read with fewer conditions:

```go
if !hasResolver {
   // use a zero value resolver with no timeout and no subsets
   resolver = structs.ServiceResolverConfigEntry{}
}
cluster, err = s.makeGatewayClusterWithConnectTimeout(cfgSnap, clusterName, &svc, resolver.ConnectTimeout)
if err != nil {
    return err
}
for subsetName := range resolver.Subsets {
    ...
}
...

```

Reply:
I like how this avoids checking `hasResolver` twice. Added it.